### PR TITLE
Generate entire catalogue client

### PR DESCRIPTION
**Pull request by michpohl**
This PR tweaks openapi-generator to generate a `Catalogue` class that serves the generated API classes and uses retrofit and our own authentication. 

The generated client is fully functional.

You can try the generation yourself by running `./catalogue/bin/generate.sh`

This is only a small step away from dynamically generating TIDAL-compatible Android clients for any submitted OpenApi-compatible spec file.

### CODE REVIEW

File: catalogue/openapi-config/openapi-config.yml [hunk 1]
```diff
@@ -8,3 +9,7 @@ additionalProperties:
   serializationLibrary: kotlinx_serialization
   supportAndroidApiLevel25AndBelow: true
   useCoroutines: true
+files:
+  Catalogue.kt.mustache:
+    templateType: SupportingFiles
+    destinationFilename: src/main/kotlin/com/tidal/sdk/catalogue/generated/Catalogue.kt
```
Comment (stoyicker) [inline catalogue/openapi-config/openapi-config.yml:15]: generated code should not be here imho. check out where dagger places generated files.

**Comment by stoyicker** [comment]
> You can try the generation yourself by running ./catalogue/bin/generate.sh

I think we should run this as part of the build process prior to compiling, rather than relying on people/CI manually calling a script when appropriate.

**Comment by michpohl** [comment]
> > You can try the generation yourself by running ./catalogue/bin/generate.sh
> 
> I think we should run this as part of the build process prior to compiling, rather than relying on people/CI manually calling a script when appropriate.

Of course, but we're at the beginning of this process. There will be a number of intermediate steps that work but not like desired in a final product.

File: catalogue/openapi-config/openapi-config.yml (hunk 1 above)
Comment (michpohl) [reply catalogue/openapi-config/openapi-config.yml:15]: I'll read up on the conventions

**Pull request closed without merging**